src/azurelinux/3.0/crossdeps/Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM azurelinuxpreview.azurecr.io/public/azurelinux/base/core:3.0
+
+RUN tdnf update -y && \
+    tdnf install -y \
+        # Provides 'su', required by Azure DevOps
+        ca-certificates \
+        git \
+        sudo \
+        util-linux \
+        wget \
+        # Common runtime build dependencies
+        awk \
+        cmake \
+        diffutils \
+        icu \
+        tar \
+        zlib-devel \
+        # Crosscomponents build dependencies
+        glibc-devel \
+        kernel-headers \
+        lttng-ust-devel \
+        # Jit rolling build dependency
+        python3-pip \
+        # diagnostics build dependency
+        lldb-devel
+
+# Validate checksums with keyring after https://github.com/microsoft/azurelinux/issues/3142 is resolved
+ENV NODE_VERSION=20.11.1
+RUN mkdir /usr/share/node && \
+    curl -O https://nodejs.org/dist/v${NODE_VERSION}/SHASUMS256.txt && \
+    curl -O -sSL https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}-linux-x64.tar.xz && \
+    grep node-v${NODE_VERSION}-linux-x64.tar.xz SHASUMS256.txt | sha256sum -c - && \
+    tar xf node-v${NODE_VERSION}-linux-x64.tar.xz -J -C /usr/share/node --strip-components=1 && \
+    ln -s /usr/share/node/bin/node /usr/local/bin/node && \
+    ln -s /usr/share/node/bin/npm /usr/local/bin/npm && \
+    ln -s /usr/share/node/bin/npx /usr/local/bin/npx && \
+    rm SHASUMS256.txt && \
+    rm node-v${NODE_VERSION}-linux-x64.tar.xz
```

Comment:
Azure Linux 3.0 should have node 20 available in the `nodejs` package so we shouldn't have to manually download it.

Reply:
That'd sure be nice.

Reply:
I see this version: `20.10.0-1.azl3`

Reply:
That should be fine. We only needed node20 for some AzDO tasks and for aspnetcore (in the VMR build).

Reply:
This is where we should start to partition stuff.

Does ASP.NET Core need to use these images? These images should be EXCLUSIVELY used runtime and other teams who have a native build (with overlapping requirements). We need to break teams using these images who have managed only builds and not accept any of their requests for changes.

I'm going to make a managed only image for teams. It isn't entirely obvious what should be in there yet.

Does `runtime` need those AzDo tasks or is that for other repos/teams?

Reply:
I think runtime needs it for Component Governance, but I'm not positive.

aspnetcore wouldn't be using this image except for in the VMR. We could have separate images for the VMR builds that include the dependencies for all constituent projects if we wanted to keep the separation.

Reply:
Ah yes, VMR. Sven and I talked about that. OK, we probably need it then. Reality is often annoying.

My point about moving repo (non-VMR) builds to another (MUCH) smaller image still stand.

Reply:
Does aspnetcore run any official builds on non-Windows platforms?

> Reality is often annoying.

Right. The current high-level breakdown should be like this:
- If the repo is managed-only, it should not need to run official builds outside Windows at all. Our builds are Windows centric for historic reasons. An exception are source build sanity checks that run on [ubuntu image in CI and 1es-mariner-2 image in internal builds](https://github.com/dotnet/arcade/blob/ce6cb9b392e6a534ffab4e8d9d2dbadcc5491b5a/eng/common/templates-official/job/source-build.yml#L49-L56).
- If the repo builds native components (are there any repos like that outside runtime?), it needs more or less the same image as runtime for official or CI build.

Reply:
With the VMR, we'll now be running official builds for each constituent repo on each platform (as part of the VMR's build command).

Outside the VMR, I believe ASP.NET Core runs the builds that produce the non-Windows runtime packs on those non-Windows platforms. I believe they do so as it matches the innerloop dev experience well and then they don't have to build custom infrastructure for "build on Windows for non-Windows" into their repo (as they do have some Windows-only assets)

Reply:
I started this "spec" to get us going on a plan. There are a TON of patterns in pipelines today. We need to reduce those a fair bit. Some are quite specialized, like in `dotnet/installer` but others seem needlessly DIY.

https://github.com/dotnet/dotnet-buildtools-prereqs-docker/issues/997